www/js/services/commHelper.ts

```diff
@@ -5,17 +5,18 @@ import { TimestampRange } from '../types/diaryTypes';
 
 /**
  * @param url URL endpoint for the request
+ * @param fetchOpts (optional) options for the fetch request. If 'cache' is set to 'reload', the cache will be ignored
  * @returns Promise of the fetched response (as text) or cached text from local storage
  */
-export async function fetchUrlCached(url) {
+export async function fetchUrlCached(url: string, fetchOpts?: RequestInit) {
   const stored = localStorage.getItem(url);
-  if (stored) {
+  if (stored && fetchOpts?.cache != 'reload') {
     logDebug(`fetchUrlCached: found cached data for url ${url}, returning`);
     return Promise.resolve(stored);
   }
   try {
-    logDebug(`fetchUrlCached: found no cached data for url ${url}, fetching`);
-    const response = await fetch(url);
+    logDebug(`fetchUrlCached: cache had ${stored} for url ${url}, not using; fetching`);
+    const response = await fetch(url, fetchOpts);
```

Comment:
Just for my own understanding, we also cache separately in the localStorage. This is just to ensure that the HTTP cache (if any) also does not return obsolete values. The log statement prints the value from the `localStorage` cache.

Reply:
Correct